ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -312,11 +316,28 @@ public static String getSqlTypeName(Schema schema) {
                + ","
                + getSqlTypeName(schema.valueSchema())
                + ">";
+      case STRUCT:
+        return getStructString(schema);
       default:
         throw new KsqlException(String.format("Invalid type in schema: %s.", schema.toString()));
     }
   }
 
+  private static String getStructString(Schema schema) {
+    StringBuilder structString = new StringBuilder("STRUCT <");
+    boolean addComma = false;
```

Comment:
This `addComma` stuff is everywhere through the code. In this class alone there are 4 occurrences of essentially the same thing. It would be good to just have one of these. Also can we not replace it with sth like:
```
return String.join(", ", 
        schema.fields()
        .stream()
        .map(f -> f.name() + " " + getSqlTypeName(f.schema()))
        .collect(Collectors.toList())) + ">";
```

Reply:
@dguy just to clarify you do NOT want to replace it with something like the code you wrote or you do want?

Reply:
I would suggest use of a better collector:

```
return schema.fields().stream()
        .map(f -> f.name() + " " + getSqlTypeName(f.schema()))
        .collect(Collectors.joining(",", "STRUCT <", ">")));
```

Reply:
@big-andy-coates I already changed it to exactly what you are suggesting! :)